### PR TITLE
Finger Curl Calculation Addition

### PR DESCRIPTION
**Pull request by chetu3319**
Adding functions to calculate the curl of fingers. 
Example usage: 
```
// Compute the curl of Index Finger of Right Hand
float indexCurl =  HandPoseUtils.IndexCurl(Handedness.Right);
```

Output ranges from 0 (No-Curl) to 1 (Fully-Curled) finger.

### CODE REVIEW

File: Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class FingerCurlExample : MonoBehaviour
+{
+    public Handedness handedness;
+
+    public GameObject indexFinger;
+    public GameObject middleFinger;
+    public GameObject ringFinger;
+    public GameObject pinkyFinger;
+
+
```
Comment (provencher) [inline Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs:47]: please remove some of the white space here.

File: Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Utils/HandPoseUtils.cs [hunk 2]
```diff
@@ -90,5 +94,98 @@ public static bool IsThumbGrabbing(Handedness hand)
             }
             return false;
         }
+
+        /*
+        * Finger Curl Utils: Util Functions to calculate the curl of a specific finger. 
+        * Author: Chaitanya Shah
+        * github: https://github.com/chetu3319
+        */
+
```
Comment (provencher) [inline Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Utils/HandPoseUtils.cs]: Please remove one white space here. Try not to keep more than one blank line before spaced out areas here.

File: Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using prvncher.MixedReality.Toolkit.Utils;
+using Microsoft.MixedReality.Toolkit.Utilities;
+
+public class FingerCurlExample : MonoBehaviour
+{
+    public Handedness handedness;
+
+    public GameObject indexFinger;
```
Comment (provencher) [inline Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs]: My preference is to have private serialized fields.

File: Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using prvncher.MixedReality.Toolkit.Utils;
+using Microsoft.MixedReality.Toolkit.Utilities;
+
+public class FingerCurlExample : MonoBehaviour
```
Comment (provencher) [inline Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs]: Please add a namespace for this class. See the other debug sample classes

File: Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
```
Comment (provencher) [inline Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs]: Please add a header here spelling out the license and such. See other classes in this project for a reference.

**Review by provencher — CHANGES_REQUESTED** [review]
Code looks great! Some minor changes.
If you can post a gif or a screenshot of what the scene changes look like, it would be great.

File: Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Utils/HandPoseUtils.cs [hunk 6]
```diff
@@ -90,5 +94,98 @@ public static bool IsThumbGrabbing(Handedness hand)
...
+        }
+
+
+        // Curl calculation of a finger based on the angle made by vectors wristToFingerKuncle and fingerKuckleToFingerTip.
+        static private float CalculateCurl(Vector3 wristJoint, Vector3 fingerKnuckleJoint, Vector3 fingerTipJoint)
+        {
+            var palmToFinger = (fingerKnuckleJoint - wristJoint).normalized;
+            var fingerKnuckleToTip = (fingerKnuckleJoint - fingerTipJoint).normalized;
+
+            var curl = Vector3.Dot(fingerKnuckleToTip, palmToFinger);
```
Comment (provencher) [inline Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Utils/HandPoseUtils.cs:174]: Clever approach to do a dot product here!

**Comment by chetu3319** [comment]
I've made the suggested updates of calculating `gripStrength` from the finger's curl value. 
I carried two different calculation: 
1. Only considering the index and middle finger's curl value. Works for full hand grip, but provides a false visual cue when only that two fingers are curled. 
2. Considering all four fingers' curl value. Works perfectly providing intended visual cue.

**Review by provencher — APPROVED** [review]

**Pull request merged**